### ISSUE TITLE
RAAS Graph Duplicating some capture zone clusters

### ISSUE THREAD

**Issue by Shanomac99** [Squad-Wiki/squad-wiki-pipeline-map-data#67]
RAAS Graph Duplicating some capture zone clusters.

![image](https://github.com/Squad-Wiki/squad-wiki-pipeline-map-data/assets/11017547/30bf98ea-2e58-4370-9674-de8908a72ef8)

**Comment by Shanomac99** [comment]
Looks like this map has multiple extra nodes attached into its RAAS Initializer. Detecting the last main point is an option for manually fixing this.

![image](https://github.com/Squad-Wiki/squad-wiki-pipeline-map-data/assets/11017547/0e6af1e6-be50-4d4f-9fc5-961da8efcd6a)

**Comment by Shanomac99** [comment]
Lashkar Valley v4 has an issue where it's possible to choose from 01 to 02a or 02b.

![image](https://github.com/Squad-Wiki/squad-wiki-pipeline-map-data/assets/11017547/82e3dc70-7e66-4bd4-ab20-8a0b81a9b195)

**Comment by Shanomac99** [comment]
pointsOrder is _not_ able to handle this in the current format 
![image](https://github.com/Squad-Wiki/squad-wiki-pipeline-map-data/assets/11017547/7f7c4482-916a-4b82-80ca-dbeeca449b53)


You cannot have two clusters occupy the same spot

**Comment by w4rum** [comment]
I encountered this same issue back when I wrote the path traversal logic for Squadlanes.

**TL;DR**: You need to extract the `DesignOutgoingLinks` array, i.e., the A->B connections, if you want to accurately represent the new RAAS layers.

Longer explanation:

The "new" layers, e.g., Narva RAAS v4, use a more complex RAAS graph. On these layers, you are no longer able to just use a simple ordering, i.e., just assign a number to each cluster.

Instead, the clusters now form a *directed acyclic graph*, with the clusters being the vertices and the links being the edges. This is different from the old layers, where each lane is a simple one-track path through the clusters.

In simpler terms:
**On old layers**, each lane always takes the same path through the clusters. On Narva RAAS v1, there are three lanes: `Center`, `East`, and `West`.

`Center` always goes through the clusters in the following order:
- `00-Team1Main`
- `01C-BP_CaptureZoneCluster`
- `02C-BP_CaptureZoneCluster`
- `03C-BP_CaptureZoneCluster`
- `04C-BP_CaptureZoneCluster`
- `05C-BP_CaptureZoneCluster`
- `100-Team2Main`

`East` and `West` work the same.

But **on the new layers**, there is only a single lane. However, that single lane can take multiple different paths.

Narva RAAS v4 only has a single lane, which uses the cluster as follows:
- `00-Team1Main_0`
- `01WCE-BP_CaptureZoneCluster`
-  `01WCE-BP_CaptureZoneCluster` can go to
    - `02W-BP_CaptureZoneCluster_4`, or
    - `02C-BP_CaptureZoneCluster_0`, or
    - `02E-BP_CaptureZoneCluster_8`
- `02W-BP_CaptureZoneCluster_4` can go to
    - `03W-BP_CaptureZoneCluster_5` or
    - `03C-BP_CaptureZoneCluster_1`
- `02C-BP_CaptureZoneCluster_0` can go to
    - `03C-BP_CaptureZoneCluster_1`, or
    - `03W-BP_CaptureZoneCluster_5`, or
    - `03E-BP_CaptureZoneCluster_9`
- ...

As you can see, there are now quite a few branches and that allows a ton of possible paths. Just sorting the clusters in a simple array can no longer represent a layer accurately. It is even possible that a layer has multiple different lengths based on which path is taken. Example:

Kohat RAAS v5 can go:
- (Length 5) `Upper Nasrat` -> `Lower Nasrat` -> `Jouzara Proper` -> `Hangu Proper` -> `Bannu Rd`
- (Length 6) `Upper Nasrat` -> `Upper Thirah` -> `Chakar Kot` -> `Suni Khel` -> `Upper Sarozai` -> `Togh Sarai`
- (Length 7)  `Upper Nasrat` -> `Upper Khadizai` -> `Ambar Banda` -> `Mohd Zai` -> `Bahadur` -> `Sarozai Highlands` -> `Togh Sarai`

On Squadlanes, I am able to represent such cases because the frontend interprets the clusters and links as a directed acyclic graph and then makes a couple of breadth-first searches to determine which points are possible to reach given the currently known points.

In order to do that, I had to change my data mining setup to also extract the `DesignOutgoingLinks` property of the `SQGraphRAASInitializerComponent`.

Example of the data mining output: https://github.com/w4rum/squadlanes/blob/master/src/assets/raas-data.yaml#L31956

**Comment by Shanomac99** [comment]
This _should_ be fixed.  Introduced links with v6.0. It's up to the user to generate the paths. Closing for now.
